src/components/Home/WalletCard.tsx

```diff
@@ -0,0 +1,82 @@
+import Link from 'next/link'
+import { formatUnits } from 'viem'
+import PeanutWalletIcon from '@/assets/icons/peanut-wallet.png'
+import Icon from '@/components/Global/Icon'
+import { shortenAddressLong } from '@/utils'
+import { IWallet } from '@/interfaces'
+
+interface WalletCardProps {
+    type: 'wallet' | 'add'
+    wallet?: IWallet
+    username?: string
+    selected?: boolean
+    onClick?: () => void
+}
+
+export function WalletCard({ type, wallet, username, selected = false, onClick }: WalletCardProps) {
+    if (type === 'add') {
+        return (
+            <motion.div className="h-full">
+                <Card
+                    className="h-full min-w-[300px] rounded-md text-black hover:cursor-pointer"
+                    shadowSize="6"
+                    onClick={onClick}
+                >
+                    <Link href="/setup" className="h-full">
+                        <Card.Content className="flex h-full flex-col items-center justify-center gap-6 p-6">
+                            <p className="text-center text-xl font-bold">Add your own ETH wallet</p>
+                            <div className="flex flex-row items-center gap-3">
+                                <Icon name="plus-circle" className="h-6 w-6" />
+                                <span className="text-base">Add BYOW wallet</span>
+                            </div>
+                        </Card.Content>
+                    </Link>
+                </Card>
+            </motion.div>
+        )
+    }
+
+    if (!wallet) return null
+
+    return (
+        <motion.div
+            className={classNames('mr-4 h-full', {
+                'opacity-40': !selected,
+            })}
+            onClick={onClick}
+            transition={{ type: 'spring', stiffness: 300, damping: 20 }}
+        >
+            <Card
+                className={classNames(
+                    'flex h-full w-[300px] flex-col gap-4 rounded-md text-white hover:cursor-pointer',
+                    {
+                        'bg-green-1': wallet.connected,
+                        'bg-purple-1': !wallet.connected,
+                    }
+                )}
+                shadowSize="6"
+            >
+                <Card.Content className="flex h-full flex-col justify-between gap-2">
+                    <div className="flex flex-row items-center gap-4">
+                        <Image src={PeanutWalletIcon} alt="" />
+                        <p className="text-md">
+                            peanut.me/<span className="font-bold">{username}</span>
+                        </p>
+                    </div>
+                    <p className="text-4xl font-black sm:text-5xl">
+                        $ {Number(formatUnits(wallet.balance, 6)).toFixed(2)}
+                    </p>
+                    <div>
+                        <div className="flex flex-col">
+                            <p className="text-xl font-black sm:text-2xl">{shortenAddressLong(wallet.address)}</p>
```

Comment:
should this be shortened, or AddressLink?

Reply:
Hmmm AddressLink doesn't make much sense here, but maybe show full address or have a way of copying the address would be good.